llama/model.py

```diff
@@ -92,46 +96,65 @@ def __init__(self, args: ModelArgs, world_size: Optional[int] = None,
 
         init_method = lambda x: x
 
-        self.wq = ColumnParallelLinear(
-            args.dim,
-            args.n_heads * self.head_dim,
-            bias=False,
-            gather_output=False,
-            init_method=init_method,
-            world_size=world_size,
-            rank=rank,
-            groups=groups,
-        )
-        self.wk = ColumnParallelLinear(
-            args.dim,
-            args.n_heads * self.head_dim,
-            bias=False,
-            gather_output=False,
-            init_method=init_method,
-            world_size=world_size,
-            rank=rank,
-            groups=groups,
-        )
-        self.wv = ColumnParallelLinear(
-            args.dim,
-            args.n_heads * self.head_dim,
-            bias=False,
-            gather_output=False,
-            init_method=init_method,
-            world_size=world_size,
-            rank=rank,
-            groups=groups,
-        )
-        self.wo = RowParallelLinear(
-            args.n_heads * self.head_dim,
-            args.dim,
-            bias=False,
-            input_is_parallel=True,
-            init_method=init_method,
-            world_size=world_size,
-            rank=rank,
-            groups=groups,
-        )
+        if args.quant:
+            self.wq = ColumnParallelLinearQuant(
```

Comment:
Can the class names be parameterized to reduce code replication?

Reply:
@wonjoolee95 mentioned this as well : ) Looks like this is a strong demand. The intension was to keep the 2 flow totally separated. Also usually quantized layer is a separated one: e.g. https://github.com/pytorch/pytorch/blob/main/torch/ao/nn/quantized/modules/linear.py

Let me parameterize it to make it cleaner.